src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenReadOnlySpanConstructionTest.cs

```diff
@@ -908,5 +915,296 @@ .maxstack  4
+                yield return new object[] { withCtor, "MyColor_UInt32", "MyColor_UInt32.Red, MyColor_UInt32.Blue", "ldind.u4", "long <PrivateImplementationDetails>.1B03AB083D0FB41E44D480F48D5BBA181C623C0594BDA1AA8EA71A3B67DBF3B14" };
+                yield return new object[] { withCtor, "MyColor_Int64", "MyColor_Int64.Red, MyColor_Int64.Blue", "ldind.i8", "<PrivateImplementationDetails>.__StaticArrayInitTypeSize=16 <PrivateImplementationDetails>.F7548C023E431138B11357593F5CCEB9DD35EB0B0A2041F0B1560212EEB6F13E8" };
+                yield return new object[] { withCtor, "MyColor_UInt64", "MyColor_UInt64.Red, MyColor_UInt64.Blue", "ldind.i8", "<PrivateImplementationDetails>.__StaticArrayInitTypeSize=16 <PrivateImplementationDetails>.F7548C023E431138B11357593F5CCEB9DD35EB0B0A2041F0B1560212EEB6F13E8" };
+            }
+        }
+
+        public static IEnumerable<object[]> CreateSpanIsUsedForSpanOfNonSize1Types_HasCreateSpan_MemberData() =>
+            from row in CreateSpan_NonSize1Type_MemberData()
+            select new object[] { row[0], row[1], row[2], row[4] };
+
+        [Theory]
+        [MemberData(nameof(CreateSpanIsUsedForSpanOfNonSize1Types_HasCreateSpan_MemberData))]
+        public void CreateSpanIsUsedForSpanOfNonSize1Types_HasCreateSpan(bool withCtor, string type, string args, string ldtokenArg)
+        {
+            string csharp = RuntimeHelpersCreateSpan + @"
+public enum MyColor_Byte : byte { Red, Orange, Yellow, Green, Blue }
+public enum MyColor_SByte : sbyte { Red, Orange, Yellow, Green, Blue }
+public enum MyColor_UInt16 : ushort { Red, Orange, Yellow, Green, Blue }
+public enum MyColor_Int16 : short { Red, Orange, Yellow, Green, Blue }
+public enum MyColor_UInt32 : uint { Red, Orange, Yellow, Green, Blue }
+public enum MyColor_Int32 : int { Red, Orange, Yellow, Green, Blue }
+public enum MyColor_UInt64 : ulong { Red, Orange, Yellow, Green, Blue }
+public enum MyColor_Int64 : long { Red, Orange, Yellow, Green, Blue }
+
+public class Test
+{";
+            csharp += withCtor ?
+                $@"    public static System.ReadOnlySpan<{type}> StaticData => new System.ReadOnlySpan<{type}>(new {type}[] {{ {args} }});" :
+                $@"    public static System.ReadOnlySpan<{type}> StaticData => new {type}[] {{ {args} }};";
+            csharp += "}";
+
+            var compilation = CreateCompilationWithMscorlibAndSpan(csharp, options: TestOptions.UnsafeReleaseDll);
+            var verifier = CompileAndVerify(compilation, verify: Verification.Skipped);
+            verifier.VerifyIL("Test.StaticData.get", @$"{{
+  // Code size       11 (0xb)
+  .maxstack  1
+  IL_0000:  ldtoken    ""{ldtokenArg}""
+  IL_0005:  call       ""System.ReadOnlySpan<{type}> System.Runtime.CompilerServices.RuntimeHelpers.CreateSpan<{type}>(System.RuntimeFieldHandle)""
```

Comment:
This test covers all the positive types intended for the optimization. Consider adding a similar test with a type that can be used for constants but would not trigger the CreateSpan optimization (such as `decimal`).

Reply:
`decimal` is covered by the CreateSpanIsNotUsedForNonSupportedTypes later on.